test/stream.test.ts

```diff
@@ -317,6 +363,20 @@ describe('DataAndMoneyStream', function () {
   })
 
   describe('end', function () {
+    it('should end without opening connection on other side if no data send', function (done) {
+      const serverConnect = sinon.spy()
+      const clientEnd = sinon.spy()
+      this.serverConn.on('stream', serverConnect)
+      const stream = this.clientConn.createStream()
+      stream.on('end', () => {
+        clientEnd()
+        assert.notCalled(serverConnect)
```

Comment:
It might, normally it happens right away before the end is called if we pass in data via end('hello') for example. Ill add an await setImmediate before we call the assertion just to be sure that is not happening.